### PR TITLE
 FISH-171 Support for multi HTTPAuthenticationMechanism

### PR DESCRIPTION
**Pull request by MeroRai**

### CODE REVIEW

File: impl/src/main/java/org/glassfish/soteria/mechanisms/jaspic/HttpBridgeServerAuthModule.java [hunk 1]
```diff
@@ -73,14 +77,33 @@
...
+
+    private void initMappings() throws ClassNotFoundException {
+        mappings.put("Basic", BasicAuthenticationMechanism.class);
+        mappings.put("Form", FormAuthenticationMechanism.class);
+        mappings.put("CustomForm", CustomFormAuthenticationMechanism.class);
+        mappings.put("JWT", Thread.currentThread().getContextClassLoader().loadClass("fish.payara.microprofile.jwtauth.eesecurity.JWTAuthenticationMechanism"));
+        mappings.put("Certificate", Thread.currentThread().getContextClassLoader().loadClass("fish.payara.security.realm.mechanisms.CertificateAuthenticationMechanism"));
+        mappings.put("Azure", Thread.currentThread().getContextClassLoader().loadClass("fish.payara.security.openid.azure.AzureOpenIdAuthenticationMechanism"));
+        mappings.put("Google", Thread.currentThread().getContextClassLoader().loadClass("fish.payara.security.openid.google.GoogleOpenIdAuthenticationMechanism"));
+        mappings.put("OAuth2", Thread.currentThread().getContextClassLoader().loadClass("fish.payara.security.oauth2.OAuth2AuthenticationMechanism"));
```
Comment (jGauravGupta) [inline impl/src/main/java/org/glassfish/soteria/mechanisms/jaspic/HttpBridgeServerAuthModule.java:104]: **Missing OpenIdAuthenticationMechanism:**
`mappings.put("OIDC", Thread.currentThread().getContextClassLoader().loadClass("fish.payara.security.openid.OpenIdAuthenticationMechanism"));`

Comment (jGauravGupta) [reply impl/src/main/java/org/glassfish/soteria/mechanisms/jaspic/HttpBridgeServerAuthModule.java:104]: Also missing the `fish.payara.security.authentication.twoIdentityStore.TwoIdentityStoreAuthenticationMechanism`

**Review by jGauravGupta — APPROVED** [review]
Patch LGTM

**Review by rdebusscher — APPROVED** [review]

**Comment by smillidge** [comment]
Is this getting pushed upstream to the Eclipse project?

**Comment by rdebusscher** [comment]
@smillidge  There is an issue open for this on the Security API and Soteria projects.

Need to discuss this at Jakarta level and see if our solution can be generalised or not.

**Comment by smillidge** [comment]
Have we modified the api? I thought we just modified Soteria?

**Comment by rdebusscher** [comment]
Just Soteria, but there is an enhancement request for the api to allow multiple http mechanism

**Pull request merged**